front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/InfrastructureDefinitionsController.groovy

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.front50.controllers
+
+import com.netflix.spinnaker.front50.exception.NotFoundException
+import com.netflix.spinnaker.front50.model.infrastructureDefinition.InfrastructureDefinition
+import com.netflix.spinnaker.front50.model.infrastructureDefinition.InfrastructureDefinitionDAO
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
+import org.springframework.web.bind.annotation.ExceptionHandler
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestMethod
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+
+import static org.springframework.http.HttpStatus.UNPROCESSABLE_ENTITY
+
+@RestController
+@RequestMapping("/infrastructure")
+@ConditionalOnExpression('${spinnaker.gcs.enabled:false} || ${spinnaker.s3.enabled:false}')
+class InfrastructureDefinitionsController {
+
+    @Autowired
+    InfrastructureDefinitionDAO infrastructureDefinitionDAO
+
+    @RequestMapping(value = "/{application:.+}/{account:.+}/history", method = RequestMethod.GET)
+    Collection<InfrastructureDefinition> getHistory(@PathVariable String application,
```

Comment:
If we switch away from using a UUID to the application + account ... than this endpoint would become:
```
/infrastructure/{infrastructureId}                    --> Get me the most recent snapshot
/infrastructure/{infrastructureId}/history       --> Get me the last N snapshots (as per `limit`)
```

I think that fits a little better with where the APIs in Front50 should be headed.